integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/connector/test/MediationLibraryServiceTestCase.java

```diff
@@ -0,0 +1,286 @@
+        Assert.assertTrue(checkAvailabilityInImports(TEMP_CONNECTOR_LIB_QNAME),
+                          "Import is not available after add action.");
+
+        //get import
+        Assert.assertEquals(getImport(HELLO_CONNECTOR_LIB_QNAME), expectedImport,
+                            "Received synapse configuration for hello-connector is incorrect.");
+
+        try {
+            getImport(INVALID_CONNECTOR);
+            Assert.fail("Get import fails for an invalid connector.");
+        } catch (AxisFault e) {
+            Assert.assertEquals(e.getMessage(), "Library Import null does not exist");
+        }
+
+        //delete import
+        deleteImport(TEMP_CONNECTOR_LIB_QNAME);
+        Assert.assertFalse(checkAvailabilityInImports(TEMP_CONNECTOR_LIB_QNAME),
+                           "Import is still available after delete action.");
+    }
+
+    /**
+     * Test for library information.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test retrieving library information.")
+    public void libraryInfoTest() throws Exception {
+
+        LibraryInfo[] allLibraryInfo = getAllLibraryInfo();
+        LibraryInfo helloLibraryInfo = null;
+
+        if (null != allLibraryInfo && allLibraryInfo.length > 0) {
+            for (int i = 0; i < allLibraryInfo.length; i++) {
+                if (allLibraryInfo[i].getLibName().equals(HELLO_LIB_NAME)) {
+                    helloLibraryInfo = allLibraryInfo[i];
+                    break;
+                }
+            }
+        }
+
+        if (null != helloLibraryInfo) {
+            Assert.assertEquals(helloLibraryInfo.getDescription(), "WSO2 ESB Custom Connector Library Hello.",
+                                "Recieved description for hello connector is incorrect.");
+            Assert.assertEquals(helloLibraryInfo.getQName(), HELLO_CONNECTOR_LIB_QNAME,
+                                "Recieved QName for hello connector is incorrect.");
+        } else {
+            Assert.fail("Hello connector does not exist in library list.");
+        }
+
+        LibraryInfo libraryInfo = getLibraryInfo(HELLO_LIB_NAME, PACKAGE_NAME);
+        if (null != libraryInfo) {
+            Assert.assertEquals(helloLibraryInfo.getPackageName(), PACKAGE_NAME,
+                                "Recieved package name for hello connector is incorrect.");
+            Assert.assertTrue(helloLibraryInfo.getStatus(), "Recieved status for hello connector is incorrect.");
+        } else {
+            Assert.fail("Failure at retrieving hello library information.");
+        }
+    }
+
+    /**
+     * Test for list of available libraries.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test retrieving libraries list.")
+    public void getLibrariesTest() throws Exception {
+
+        String[] libraries = getAllLibraries();
+        boolean isHelloExists = false;
+
+        if (null != libraries && libraries.length > 0) {
```

Comment:
Can't we use .isBlank() here?

Reply:
isBlank can be used for a String value. Since libraries is a String array, we cannot use that.